public/ngapp/register/controllers.js

```diff
@@ -204,21 +205,33 @@ controllers.controller("RoutersCtrl",["$scope", '$rootScope', '$state', 'routerF
       }, 3000);
     };
 
+    $scope.filterData = function(internal){
+      var filteredHost = [];
+      $scope.filteredData = angular.copy($scope.data);
+      _.each($scope.filteredData, function(data){
+        filteredHost = _.filter(data.Router.Host, function(host){
+          return host.Internal == internal;
+        });
+        data.Router.Host = filteredHost;
+      });
+      $scope.currentData = $scope.filteredData;
+    };
+
     $scope.addRouter = function(currentZone, currentHost, Internal){
-      var templateUrl = 'ngapp/templates/addModal.html',name = currentHost,
-      itemType = "router";
+      var templateUrl = 'ngapp/templates/addModal.html', name = currentHost,
+      itemType = "router", router, host;
```

Comment:
One last change related to readability.
Align `itemType` with `templateUrl`. Update variables of other functions too.
Also, after this variable declarations given one blank line.